drivers/can/can_mcux_mcan.c

```diff
@@ -169,14 +130,14 @@ static const struct can_driver_api mcux_mcan_driver_api = {
 	};								\
 									\
 	static const struct can_mcan_config can_mcan_config_##n =	\
-		MCUX_MCAN_MCAN_INIT(n);					\
+		CAN_MCAN_CONFIG_INITIALIZER(DT_DRV_INST(n),		\
+					    &mcux_mcan_config_##n);	\
```

Comment:
maybe provide DT based macros to avoid usage of `DT_DRV_INST`, maybe to define as well, e.g.

```c
CAN_MCAN_CONFIG_DT_INST_DEFINE(n, &mcux_can_config_##n);
```

Reply:
Good idea. I had initially added `inst` variants as the only variants, but decided to go for a more generic `node_id` approach. I have added both now and (re-)named them to reflect the DT part.